src/sessions/VirtualPageLoadTimer.ts

```diff
@@ -168,7 +168,7 @@ export class VirtualPageLoadTimer extends MonkeyPatched {
             .catch((error) => {
                 throw error;
             })
-            .finally(self.decrementFetchCounter());
+            .finally(self.decrementFetchCounter.bind(self));
```

Comment:
basically we want to only call the `decrementFetchCounter` when `finally` is invoked and not when the wrapping function is called.

Reply:
Could you add a unit test that covers the behavior?

Reply:
sure

Reply:
Done